tools/codegen.js

```diff
@@ -790,6 +790,20 @@
     if (type === 'boolean') {
       return value.toString();
     }
+    if (type === 'number'
+        && node
+        && node.startToken
+        && node.startToken.type === 'Numeric'
+        && node.startToken.value.length > 1
+        && node.startToken.value.startsWith('0')) {
```

Comment:
I think we can use `node.raw` instead of `node.startToken.value` (based on my experimenting [here](https://astexplorer.net)). Let me know if I'm wrong.
```suggestion
    if (type === 'number'
        && node
        && node.raw.length > 1
        && node.raw.startsWith('0')) {
```